packages/grafana-ui/src/components/Combobox/useOptions.ts

```diff
@@ -93,16 +93,39 @@ export function useOptions<T extends string | number>(rawOptions: AsyncOptions<T
     [debouncedLoadOptions, isAsync]
   );
 
-  const finalOptions = useMemo(() => {
-    let currentOptions = [];
-    if (isAsync) {
-      currentOptions = addCustomValue(asyncOptions);
-    } else {
-      currentOptions = addCustomValue(rawOptions.filter(itemFilter(userTypedSearch)));
+  const reorganizeOptions = useCallback((options: Array<ComboboxOption<T>>) => {
```

Comment:
Maybe add a comment here that the reorganization is based on the groups.

Reply:
We added a comment about this in line 107 🤔 Do you want me to move it to here?

Reply:
Yeah, or maybe rename the variable, so that you know the reorganization happens because of groups. Then you know what the function does without looking further into it. The comment on 107 is just to clarify the order of the options.